web-api/src/personalization/routes.rs

```diff
@@ -472,20 +478,68 @@ impl SemanticSearchQuery {
     }
 }
 
+/// Represents body of a POST `semantic_search_from_text` request.
+#[derive(Debug, Clone, Deserialize)]
+struct SemanticSearchText {
+    text: String,
+    document_count: Option<usize>,
+    min_similarity: Option<f32>,
+}
+
 #[derive(Serialize)]
 struct SemanticSearchResponse {
     documents: Vec<PersonalizedDocumentData>,
 }
 
-async fn semantic_search(
+async fn semantic_search_from_text(
+    state: Data<AppState>,
+    Json(body): Json<SemanticSearchText>,
+) -> Result<impl Responder, Error> {
+    let text = body.text.trim();
+    let count = match body.document_count {
+        Some(count) => count,
+        _ => state.config.semantic_search.default_number_documents,
+    };
```

Comment:
:art: this could be shortened to
```suggestion
    let count = body
        .document_count
        .unwrap_or(state.config.semantic_search.default_number_documents);
```

:wrench: but if we want to merge this we should probably use a more robust handling for all the optional request parameters, similarly to the impls on `SemanticSearchQuery`.